hypershift-operator/controllers/nodepool/kubevirt.go

```diff
@@ -1,39 +1,167 @@
 package nodepool
 
 import (
+	"fmt"
+
 	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
+	apiresource "k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 	capikubevirt "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1alpha1"
 )
 
+const (
+	defaultMemory       = "4Gi"
+	defaultCores        = uint32(2)
+	defaultVolumeSizeGi = 16
+)
+
+func virtualMachineTemplateBase(kvPlatform *hyperv1.KubevirtNodePoolPlatform) *capikubevirt.VirtualMachineTemplateSpec {
+
+	rootVolumeName := "rhcos"
+	runAlways := kubevirtv1.RunStrategyAlways
+	memory := apiresource.MustParse(defaultMemory)
+	cores := defaultCores
+	pullMethod := v1beta1.RegistryPullNode
+
+	image := ""
+
+	volumeSize := apiresource.MustParse(fmt.Sprintf("%vGi", defaultVolumeSizeGi))
+
+	if kvPlatform.Compute.Memory != nil {
+		memory = *kvPlatform.Compute.Memory
```

Comment:
this should never be nil since we are defaulting in the API. However for robustness I'd expect this to be in a validate function which sets a failed condition if the values are empty.